mfkdf2-web/test/mfkdf2/security.test.ts

```diff
@@ -0,0 +1,843 @@
+        password1: await mfkdf.derive.factors.password('password1'),
+        password2: await mfkdf.derive.factors.password('password2')
+      })
+
+      derive.key.toString('hex').should.equal(setup.key.toString('hex'))
+    })
+
+    test('incorrect', async () => {
+      const setup = await mfkdf.policy.setup(
+        await mfkdf.policy.and(
+          await mfkdf.setup.factors.password('password1', { id: 'password1' }),
+          await mfkdf.setup.factors.password('password2', { id: 'password2' })
+        )
+      )
+
+      const derive = await mfkdf.policy.derive(
+        setup.policy,
+        {
+          password1: await mfkdf.derive.factors.password('password2'),
+          password2: await mfkdf.derive.factors.password('password1')
+        },
+        false
+      )
+
+      derive.key.toString('hex').should.not.equal(setup.key.toString('hex'))
+    })
+  })
+
+  /* TODO (@lonerapier): Implement this test after sharks repo is updated
+  suite('share-indistinguishability', () => {
+    test('share-size', async () => {
+      const secret = crypto.randomBytes(32)
+
+      const shares1 = mfkdf.secrets.share(secret, 1, 3)
+      shares1.should.have.length(3)
+      for (const share of shares1) {
+        share.should.have.length(32)
+      }
+      mfkdf.secrets
+        .combine(shares1.slice(0, 1).concat([null, null]), 1, 3)
+        .toString('hex')
+        .should.equal(secret.toString('hex'))
+      mfkdf.secrets
+        .combine([null, null].concat(shares1.slice(2, 3)), 1, 3)
+        .toString('hex')
+        .should.equal(secret.toString('hex'))
+
+      const shares2 = mfkdf.secrets.share(secret, 2, 3)
+      shares2.should.have.length(3)
+      for (const share of shares2) {
+        share.should.have.length(32)
+      }
+      mfkdf.secrets
+        .combine(shares2.slice(0, 2).concat([null]), 2, 3)
+        .toString('hex')
+        .should.equal(secret.toString('hex'))
+      mfkdf.secrets
+        .combine([null].concat(shares2.slice(1, 3)), 2, 3)
+        .toString('hex')
+        .should.equal(secret.toString('hex'))
+
+      const shares3 = mfkdf.secrets.share(secret, 3, 3)
+      shares3.should.have.length(3)
+      for (const share of shares3) {
+        share.should.have.length(32)
+      }
+      mfkdf.secrets
+        .combine(shares3.slice(0, 3), 3, 3)
+        .toString('hex')
+        .should.equal(secret.toString('hex'))
+    })
+  })
+  */
+
+  suite('share-encryption', () => {
+    test('correct', async () => {
+      const setup = await mfkdf.setup.key([
+        await mfkdf.setup.factors.password('password1', { id: 'password1' }),
+        await mfkdf.setup.factors.password('password2', { id: 'password2' })
+      ])
+
+      // TODO (@lonerapier): current ts api doesn't return closure
+      // const materialp1 = await mfkdf.derive.factors.password('password1')(
+      //   setup.policy.factors[0].params
+      // )
+      const materialp1 = await mfkdf.derive.factors.password('password1')
```

Comment:
modified this from reference implementation. particularly reference returns a closure, but bindings compute this eagerly. This usually doesn't interface with the end user.

Should this be made exactly similar to reference?

Reply:
It's probably fine, but the more exact we are now, the more reliable our testing is. We can always change it in the future.

With that said, if it's a tough change, don't worry about it.